src/dapps/saga.ts

```diff
@@ -26,10 +48,64 @@ export function* handleOpenDapp(action: PayloadAction<DappSelectedAction>) {
   }
 }
 
+export function* handleFetchDappsList() {
+  const dappsListApiUrl = yield select(dappsListApiUrlSelector)
+  const address = yield select(walletAddressSelector)
+  const language = yield select(currentLanguageSelector)
+  const shortLanguage = language.split('-')[0]
+
+  const response = yield call(
+    fetch,
+    `${dappsListApiUrl}?language=${shortLanguage}&address=${address}`,
+    {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    }
+  )
+
+  if (response.ok) {
+    try {
+      const result: {
+        applications: Application[]
+        categories: DappCategory[]
+        featured: Application
+      } = yield call([response, 'json'])
+
+      const dappsList: Dapp[] = result.applications.map((application) => {
+        return {
+          id: application.id,
+          categoryId: application.categoryId,
+          name: application.name,
+          iconUrl: application.logoUrl,
+          description: application.description,
+          dappUrl: application.url.replace('{{address}}', address ?? ''),
+          isFeatured: application.id === result.featured.id,
+        }
+      })
+
+      yield put(fetchDappsListCompleted({ dapps: dappsList, categories: result.categories }))
+    } catch (error) {
+      Logger.error(TAG, 'Could not parse dapps response', error)
+      yield put(fetchDappsListFailed({ error: 'Could not parse dapps' }))
+    }
+  } else {
+    yield put(fetchDappsListFailed({ error: 'Could not fetch dapps' }))
+  }
+}
+
 export function* watchDappSelected() {
   yield takeLatest(dappSelected.type, handleOpenDapp)
 }
 
+export function* watchFetchDappsList() {
+  yield takeEvery(fetchDappsList.type, handleFetchDappsList)
```

Comment:
Wondering if we should use [`takeLeading`](https://redux-saga.js.org/docs/api/#takeleadingpattern-saga-args).
So it only does the fetch while no other fetch is already happening.